### PR TITLE
Remove Reconciler forking

### PR DESCRIPTION
**Pull request by kassens**
We've heard from multiple contributors that the Reconciler forking mechanism was confusing and/or annoying to deal with. Since it's currently unused and there's no immediate plans to start using it again, this removes the forking.

### CODE REVIEW

**Comment by acdlite** [comment]
Rather than removing all the infra around it, can you revert just the forked files and keep the rest of the infra in place so we can enable it again in the future?

Although we're not using it currently, it's been extremely helpful several times in the past. I used it over the summer to land a complex refactor.

**Review by acdlite — APPROVED** [review]
Actually I retract my objection. If/when I want to use this again in the future, I'll implement it a different way that forks at the entire package level, rather than per-file. I think originally we did the per-file thing because it started as only forking a handful of files but it evolved into covering essentially the whole thing.

File: scripts/rollup/build.js [hunk 1]
```diff
@@ -361,9 +361,8 @@ function getPlugins(
       __UMD__: isUMDBundle ? 'true' : 'false',
       'process.env.NODE_ENV': isProduction ? "'production'" : "'development'",
       __EXPERIMENTAL__,
-      // Enable forked reconciler.
-      // NOTE: I did not put much thought into how to configure this.
-      __VARIANT__: bundle.enableNewReconciler === true,
+      // NOTE: Should only be used in test files.
+      __VARIANT__: true,
```
[the remark this answers is not in the document]
Comment (acdlite) [reply scripts/rollup/build.js:365]: Yeah I think you can remove it

File: packages/react-dom/index.classic.fb.js [hunk 2]
```diff
@@ -30,7 +30,6 @@ export {
   unstable_batchedUpdates,
   unstable_createEventHandle,
   unstable_flushControlled,
-  unstable_isNewReconciler,
```
[the remark this answers is not in the document]
Comment (acdlite) [reply packages/react-dom/index.classic.fb.js:33]: It was used in www at one point to detect which version of the reconciler was enabled at runtime. Shouldn't need it anymore.

**Comment by acdlite** [comment]
Do you mind renaming the files in a PR all by itself so that the history is preserved? And so existing PRs can be rebased cleanly. Something like:

- PR 1: Delete all the build/testing infra, and all the .old files, but keep the .new suffix on the modules.
- PR 2: Remove the .new suffix from the files.

I messed this up when I introduced the forked files originally because, although I did it in multiple commits, I only had a single PR and they got squashed when I merged it 😆

**Comment by kassens** [comment]
Step one is in #25774 now.

**Review by ikhtearalamshawonmollah54321 — COMMENTED** [review]
###

**Pull request closed without merging**